src/javascripts/modules/pillbar.js

```diff
@@ -2,52 +2,39 @@ export default class Pillbar {
   constructor(el) {
     this.el = el
     this.setVars()
-    this.setupListeners()
+    this.createObserver()
   }
 
-  setVars(el) {
-    /* Ticking essentially == isRequestingAnimationFrame; I'm leaving
-    it as 'ticking' to follow requestingAnimationFrame convention/docs */
-    this.ticking = false
-    this.lastScrollY = 0
-    this.el.pillbarOffset = this.el.offsetTop
+  setVars() {
+    this.pillbar = document.querySelector('.pillbar')
+    this.titleBox = document.querySelector('.title-box')
   }
 
-  setupListeners() {
-    window.addEventListener('resize', () => this.handleResize())
-    window.addEventListener('scroll', () => this.handleScroll())
-  }
-
-  handleScroll() {
-    this.lastScrollY = window.scrollY
-    if (!this.ticking) {
-      window.requestAnimationFrame(() => {
-        this.togglePillbar()
-      })
+  createObserver() {
+    this.options = {
+      root: null,
+      rootMargin: "0px",
+      threshold: 0,
     }
-    this.ticking = true
-  }
-
-  handleResize() {
-    clearTimeout(this.timeout)
-    this.timeout = setTimeout(() => this.getPillbarOffset(), 200)
-  }
 
-  getPillbarOffset() {
-    if(!this.el.isPinned){
-      this.el.pillbarOffset = this.el.offsetTop
-    }
+    this.observer = new IntersectionObserver(this.handleIntersect.bind(this), this.options)
+    this.observer.observe(this.titleBox)
   }
 
-  togglePillbar() {
-    this.ticking = false
-
-    if (this.lastScrollY > this.el.pillbarOffset) {
-      this.el.classList.add('pillbar-fixed')
-      this.el.isPinned = true
-    } else {
-      this.el.classList.remove('pillbar-fixed')
-      this.el.isPinned = false
-    }
+  handleIntersect(entries, observer) {
+    entries.forEach((entry) => {
+      if (entry.intersectionRatio > 0 && !entry.isRequestingAnimationFrame) {
+        window.requestAnimationFrame(() => {
```

Comment:
Can you explain why you're using `requestAnimationFrame` on top of `IntersectionObserver`?

Reply:
Since I was using requestAnimationFrame back when I was using an eventListener to track scrolling, I figured I could continue using it while using IntersectionObserver. The ultimate goal being to optimize for performance. I could be off with that line of thinking, though.

Reply:
Wondering if it's a micro-optimization that's adding complexity to your code over performance benefits. If you have time to run some experiments using dev tools I'd be fascinated to see the results.

Reply:
Yeah I'll run through some now and let you know what I find out! Did you have any ideas on a workaround for the fact that the entries array rewrites with every intersection? I can't think of another way to skip what would theoretically be entries[0], because in reality all intersections are stored as entries[0]

Reply:
I have not. In my script I'm using something like:

```javascript
myCallback(target) {
  // do stuff with `target[0]`
}
```

Reply:
I ran a couple tests on the pillbar animation and it looks like any difference is miniscule. Here are representative results:

With requestAnimationFrame:
![screen shot 2018-06-22 at 2 27 06 pm](https://user-images.githubusercontent.com/23386632/41793222-6d43717c-7629-11e8-910e-0cdfffb53c94.jpg)

Without:
![image](https://user-images.githubusercontent.com/23386632/41793208-635973dc-7629-11e8-8028-809efa377804.png)

Reply:
This is interesting and it looks like there's _some_ advantage! I ran across this article that suggests using `requestAnimationFrame` in conjunction with elements coming into view.

https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API/Timing_element_visibility

Reply:
In that case do you recommend continuing with using requestAnimationFrame?

Reply:
My inclination is to build up from the simplest implementation as needed to keep things small and maintainable. If you think it's worth it, then go ahead and add it back (sorry, since you already removed it).